File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/RankingsPollingPreferenceStore.java
package com.garpr.android.preferences;

import android.net.Uri;
import android.support.annotation.NonNull;

import com.garpr.android.models.PollFrequency;
import com.garpr.android.models.SimpleDate;
import com.garpr.android.preferences.persistent.BasePreferenceStore;

public interface RankingsPollingPreferenceStore extends BasePreferenceStore {

    @NonNull
    Preference<Boolean> getChargingRequired();

    @NonNull
    Preference<Boolean> getEnabled();

    @NonNull
    Preference<SimpleDate> getLastPoll();

    @NonNull
    Preference<PollFrequency> getPollFrequency();

    @NonNull
    Preference<SimpleDate> getRankingsDate();

    @NonNull
    Preference<Uri> getRingtone();

    @NonNull
    Preference<Boolean> getVibrationEnabled();

    @NonNull
    Preference<Boolean> getWifiRequired();

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/RankingsPollingPreferenceStore.kt
package com.garpr.android.preferences

import com.garpr.android.data.models.PollFrequency
import com.garpr.android.data.models.SimpleDate
import java.net.URI as JavaUri

interface RankingsPollingPreferenceStore : PreferenceStore {

    val chargingRequired: Preference<Boolean>

    val enabled: Preference<Boolean>

    val ringtone: Preference<JavaUri>

    val lastPoll: Preference<SimpleDate>

    val pollFrequency: Preference<PollFrequency>

    val rankingsId: Preference<String>

    val vibrationEnabled: Preference<Boolean>

    val wifiRequired: Preference<Boolean>

}
